base_import_pdf_by_template/wizards/wizard_base_import_pdf_upload.py

```diff
@@ -202,7 +201,15 @@ def _process_form(self):
                         model_name=template.child_model, related_model="lines"
                     )._process_set_value_form(line_form, field_name, line[field_name])
         try:
-            record = model_form.save()
+            # Prepare vals (similar to .save()) + apply defaults (in case it has changed
+            # in some onchange for example: warehouse_id from sale orders)
+            vals = model_form._values_to_save()
+            for key in list(ctx.keys()):
+                if key.startswith("default_"):
+                    real_key = key.replace("default_", "")
+                    if real_key in vals:
+                        vals.update({real_key: ctx[key]})
```

Comment:
Some improvements for code readability:

```suggestion
            for key in ctx:
                if key.startswith("default_"):
                    field = key.replace("default_", "")
                    if field in vals:
                        vals.update({field: ctx[key]})
```